zerver/lib/actions.py

```diff
@@ -2246,14 +2246,20 @@ def validate_sender_can_write_to_stream(sender: UserProfile,
     # Our caller is responsible for making sure that `stream` actually
     # matches the realm of the sender.
 
-    if stream.is_announcement_only:
-        if sender.is_realm_admin or is_cross_realm_bot_email(sender.delivery_email):
-            pass
-        elif sender.is_bot and (sender.bot_owner is not None and
-                                sender.bot_owner.is_realm_admin):
-            pass
-        else:
-            raise JsonableError(_("Only organization administrators can send to this stream."))
+    # Organization admins can send to any stream, irrespective of the stream_post_policy value.
+    if sender.is_realm_admin or is_cross_realm_bot_email(sender.delivery_email):
```

Comment:
Another follow-up project: We should replace `is_cross_realm_bot_email` with a new function `is_cross_realm_bot` that also validates `sender.realm.string_id == settings.SYSTEM_BOT_REALM` everywhere that it's used.